enhancements/kubelet/kubelet-node-sizing.md

```diff
@@ -0,0 +1,211 @@
+
+## Proposal
+
+### Installer Changes
+
+If the user wishes to enable `auto node sizing` during the installation of the cluster itself, they will have to indicate that using a field `metadata.autoNodeSizing` in installation configuration. e.g. 
+
+```yaml
+apiVersion: v1
+baseDomain: gcp.devcluster.openshift.com
+compute:
+- architecture: amd64
+  hyperthreading: Enabled
+  name: worker
+  platform: {}
+  replicas: 3
+controlPlane:
+  architecture: amd64
+  hyperthreading: Enabled
+  name: master
+  platform: {}
+  replicas: 3
+metadata:
+  creationTimestamp: null
+  name: testcluster1
+  autoNodeSizing: true
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineNetwork:
+  - cidr: 10.0.0.0/16
+  networkType: OpenShiftSDN
+  serviceNetwork:
+  - 172.30.0.0/16
+platform:
+  gcp:
+    projectID: openshift-gce-devel
+    region: asia-south1
+publish: External
+pullSecret: </snip>
+```
+
+### Kubelet systemd service
+
+```
+[Unit]
+Description=Kubernetes Kubelet
+Wants=rpc-statd.service network-online.target crio.service
+After=network-online.target crio.service
+
+[Service]
+Type=notify
+ExecStartPre=/bin/mkdir --parents /etc/kubernetes/manifests
+ExecStartPre=/bin/rm -f /var/lib/kubelet/cpu_manager_state
+EnvironmentFile=/etc/os-release
+EnvironmentFile=-/etc/kubernetes/kubelet-workaround
+EnvironmentFile=-/etc/kubernetes/kubelet-env
+EnvironmentFile=/etc/kubernetes/node-sizing
+
+ExecStart=/usr/bin/hyperkube \
+    kubelet \
+      --config=/etc/kubernetes/kubelet.conf \
+      --bootstrap-kubeconfig=/etc/kubernetes/kubeconfig \
+      --kubeconfig=/var/lib/kubelet/kubeconfig \
+      --container-runtime=remote \
+      --container-runtime-endpoint=/var/run/crio/crio.sock \
+      --runtime-cgroups=/system.slice/crio.service \
+      --node-labels=node-role.kubernetes.io/worker,node.openshift.io/os_id=${ID} \
+      --node-ip=${KUBELET_NODE_IP} \
+      --minimum-container-ttl-duration=6m0s \
+      --volume-plugin-dir=/etc/kubernetes/kubelet-plugins/volume/exec \
+      --cloud-provider=gce \
+      --cloud-config=/etc/kubernetes/cloud.conf \
+      --pod-infra-container-image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:9519ae9a0a3e262e311c7f12a08adb2568e29e1576d2c6c229fd5d355c551d4b \
+      --system-reserved=cpu=${CPU_RESERVED},memory=${MEMORY_RESERVED},ephemeral-storage=${EPHEMERAL_RESERVED}
+      --v=${KUBELET_LOG_LEVEL}
+
+Restart=always
+RestartSec=10
+
+[Install]
+WantedBy=multi-user.target
+```
+
+Node sizing values will be calculated by the script `/etc/kubernetes/node-sizing`. This script will set the environment variables `CPU_RESERVED`, `MEMORY_RESERVED` and `EPHEMERAL_RESERVED` which will be used for setting the kubelet parameter `system-reserved`
+
+The script `/etc/kubernetes/node-sizing` will be executed using `ExecStartPre`
```

Comment:
I did a quick audit of MCO... Let write the script to /usr/local/bin. `_base/files/configure-ovs-network.yaml` is one example that writes to /usr/local/bin that already exists.

Reply:
Thanks, updated.